contracts/DutchExchange.sol

```diff
@@ -634,125 +628,87 @@ contract DutchExchange {
     )
         public
         constant
-        returns (uint num, uint den)
+        returns (fraction price)
```

Comment:
With these changes functions that return `structs` (`getPrice` and `priceOracle`) can't be called from JavaScript:

```
invalid solidity type!: tuple
```

`(uint num, uint den)`, which was returned previously, is properly converted into `BigNumber[2]`
It would be nice to see either return to the `(uint, uint)` convention or a helper function - wrapper over `getPrice`, something like:
```
function getPriceForJS(...) public constant returns (uint, uint) {
    fraction memory price = getPrice(...);
    return (price.num, price.den);
}
```

Reply:
ran into the same issues. Will write a wrapper function, since we need the returns fraction, to save variables.

Reply:
Got help from stefan and rimeissner:
this should be possible with the new solidity version
0.4.19
https://github.com/ethereum/solidity/blob/develop/Changelog.md#0419-2017-11-30